dts/bindings/adc/adc-controller.yaml

```diff
@@ -12,3 +13,102 @@ properties:
     "#io-channel-cells":
       type: int
       required: true
+
+    "#address-cells":
+      required: false
+      const: 1
+
+    "#size-cells":
+      required: false
+      const: 0
+
+child-binding:
+    description: |
+      Channel configuration. See the adc_channel_cfg structure in adc.h.
```

Comment:
As a layering convention, we try not to mention specific APIs that might be using a binding from their descriptions, since bindings in principle are hardware description, and should therefore be independent of the APIs layered on top of them. So this information should be removed. Instead, the ADC API documentation can describe how it uses this child binding.